### ISSUE TITLE
Keep already existing nodes when connecting two nodes

### ISSUE THREAD

**Issue by hikemaniac** [openstreetmap/iD#4974]
In many of my changesets I connect newly created areas with previously existing ones. Unfortunately iD will replace a lot of older nodes with my newly created ones. This results in mass deletion and weird error messages from the editor itself (You have deleted x objects...) even though the user is pretty convinced he didn't do such a change. The older nodes are only replaced when a new node is created, but isn't yet connected to any area. When it is then dragged and connected, the older node gets replaced. Would it be possible to keep already existing nodes when connecting two of them?

**Comment by bhousel** [comment]
This is probably the same as #4674 
I can change `actionConnect` to prefer an existing node to keep as the `survivor`.

https://github.com/openstreetmap/iD/blob/c42556ec96beea2de880a04a177a68931c4b30a4/modules/actions/connect.js#L20-L21

**Comment by hikemaniac** [comment]
You are right, the situation is different, but the core issue is the same. I'll close it if you want to.

**Comment by bhousel** [comment]
Nah ok to leave open, I'm fixing it now..
